### PR TITLE
S94.16 Add API for auth manager users

### PR DESCRIPTION
**Pull request by pvaass**

### CODE REVIEW

File: Controller/AuthManagerApiController.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+App::uses('Controller', 'Controller');
+App::uses('ApiExceptionFactory', 'CakePHPUtil.Lib/Api/Exceptions');
+App::uses('JsonApiResponse', 'CakePHPUtil.Lib/Api/Response');
```
Comment (swordbeta) [inline Controller/AuthManagerApiController.php]: Unued

File: Controller/AuthManagerApiController.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	 * @var array
+	 */
+	protected $_scopesForAction = [];
+
+	/**
+	 * @var array
+	 */
+	public $components = [
+		'Auth' => [
+			'authenticate' => ['CakePHPUtil.ApiQuery'],
```
Comment (swordbeta) [inline Controller/AuthManagerApiController.php]: Is it necessary to support the JWT in the querystring here?

File: Controller/AuthManagerApiController.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+App::uses('Controller', 'Controller');
+App::uses('ApiExceptionFactory', 'CakePHPUtil.Lib/Api/Exceptions');
+App::uses('JsonApiResponse', 'CakePHPUtil.Lib/Api/Response');
+App::uses('AuthManagerApiScope', 'AuthManager.Lib/Api/Scopes');
+App::uses('ApiScopeValidator', 'CakePHPUtil.Lib/Api/Scopes');
+
+/**
+ * Class CIAApiController
```
Comment (swordbeta) [inline Controller/AuthManagerApiController.php]: Uhm?

File: Controller/MediaPlatformUsersApiController.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		return JsonApiResponse::data($response);
+	}
+
+	/**
+	 * @return MediaPlatformUsersApiScope
+	 */
+	protected function _getNewScopeClass() {
+		return new MediaPlatformUsersApiScope();
+	}
+}
```
Comment (swordbeta) [inline Controller/MediaPlatformUsersApiController.php]: Missing docblocks

File: Controller/MediaPlatformUsersApiController.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ */
+class MediaPlatformUsersApiController extends AuthManagerApiController {
+	public $uses = ['AuthManager.MediaPlatformUser'];
+
+	public function index() {
+		$mediaPlatformId = $this->request->query('media_platform');
+		$users = $this->MediaPlatformUser->listUsers($mediaPlatformId);
+
+		$response = [];
+		foreach($users as $id => $name) {
```
Comment (swordbeta) [inline Controller/MediaPlatformUsersApiController.php]: space

File: Test/Case/Controller/MediaPlatformUsersApiControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	public function testReadWithValidId() {
+		$result = json_decode($this->testAction('/AuthManager/api/1/users?media_platform=3'), true);
+		$this->assertSame(
+			[
+				['id' => 1, 'name' => 'test@test.com'],
+				['id' => 2, 'name' => 'test@test.com']
+			],
+			$result['data']
+		);
+	}
```
Comment (swordbeta) [inline Test/Case/Controller/MediaPlatformUsersApiControllerTest.php]: And what happens when you don't pass the `media_platform` querystring?

**Comment by pvaass** [comment]
@swordbeta fixed

File: Test/Case/Controller/MediaPlatformUsersApiControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+	/**
+	 *
+	 */
+	public function testWithoutMediaPlatform() {
+		$result = json_decode($this->testAction('/AuthManager/api/1/users'));
+		$this->assertEmpty($result->data);
```
Comment (swordbeta) [inline Test/Case/Controller/MediaPlatformUsersApiControllerTest.php]: Shouldn't it return all mediaplatform users?

**Comment by pvaass** [comment]
@swordbeta ok, fixed

**Comment by expand-dev** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [MediaPlatformUser.php#L135](https://github.com/ExpandOnline/AuthManager/blob/381687fc0fff2a30e58164cefda2802670a8579d/Model/MediaPlatformUser.php#L135): Remove the literal "false" boolean value. [rule](http://localhost:9000/sonarqube/coding_rules#rule_key=php%3AS1125)

File: Test/Case/Controller/MediaPlatformUsersApiControllerTest.php [hunk 8]
```diff
@@ -26,6 +26,8 @@ public function setUp() {
 	 */
 	public function testReadWithInvalidId() {
 		$result = json_decode($this->testAction('/AuthManager/api/1/users?media_platform=1123123123'));
+
+		var_dump($result->data);
```
Comment (swordbeta) [inline Test/Case/Controller/MediaPlatformUsersApiControllerTest.php]: var dump!

**Comment by expand-dev** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [MediaPlatformUser.php#L135](https://github.com/ExpandOnline/AuthManager/blob/7e9c35670325183122ec0e53d69473e5279e357d/Model/MediaPlatformUser.php#L135): Remove the literal "false" boolean value. [rule](http://localhost:9000/sonarqube/coding_rules#rule_key=php%3AS1125)

File: Test/Case/Controller/MediaPlatformUsersApiControllerTest.php (hunk 8 above)
Comment (pvaass) [inline Test/Case/Controller/MediaPlatformUsersApiControllerTest.php]: fixed

**Pull request merged**